lib/demangler-win32.js

```diff
@@ -0,0 +1,163 @@
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+"use strict";
+
+const utils = require('./utils'),
+    logger = require('./logger').logger,
+    DemanglerCPP = require("./demangler-cpp").Demangler;
+
+class DemanglerWin32 extends DemanglerCPP {
+    constructor(demanglerExe, compiler) {
+        super(demanglerExe, compiler);
+
+        // 0x28090 stands for:
+        //   - 0x00010 : Disable expansion of the declaration language specifier
+        //   - 0x00080 : Disable expansion of access specifiers for members
+        //   - 0x08000 : Disable enum/class/struct/union prefix
+        //   - 0x20000 : Disable expansion of __ptr64 keyword
+        this.flags = "0x28090";
+        this.allDecoratedLabels = /\?[a-zA-Z@$?_][a-zA-Z0-9@$?_<>]*/g;
+        this.allDecoratedLabelsWithQuotes = /"\?[a-zA-Z@$?_][a-zA-Z0-9@$?_<>]*"/;
+
+        // this is true for clang output on windows
+        // we set this to true if we see it, and false if we don't
+        // null means we haven't set it yet.
+        this.hasQuotesAroundDecoratedLabels = null;
+    }
+
+    collectLabels() {
+        this.win32RawSymbols = [];
+        for (const asmLine of this.result.asm) {
+            const labels = asmLine.text.match(this.allDecoratedLabels);
+            if (labels) {
+                if (this.hasQuotesAroundDecoratedLabels === null) {
+                    this.hasQuotesAroundDecoratedLabels =
+                        asmLine.text.match(this.allDecoratedLabelsWithQuotes) !== null;
+                }
+                for (const label of labels) {
+                    this.win32RawSymbols.push(label);
+                }
+            }
+        }
+    }
+
+    processOutput(translations) {
+        for (const asmLine of this.result.asm) {
+            const labels =
+                (this.hasQuotesAroundDecoratedLabels)
+                    ? asmLine.text.match(this.allDecoratedLabelsWithQuotes)
+                    : asmLine.text.match(this.allDecoratedLabels);
+            if (labels) {
+                let [ , beforeComment, afterComment] = asmLine.text.match(/(.*)(;.*)?/);
+                for (const label of labels) {
+                    const replacement = translations[label];
+                    if (replacement) {
+                        beforeComment = beforeComment.replace(label, replacement);
+                    } else {
+                        logger.warn("something went wrong: ", label, " doesn't have an undecoration.");
+                    }
+                }
+                asmLine.text = beforeComment + (afterComment ? afterComment : "");
```

Comment:
Could be `beforeComment + (afterComment || ")` I think? I'm just trying to look like I know how to code JS now haha (and I'm probably naïvely missing something)